Code/Framework/AzCore/AzCore/Jobs/Internal/JobTypeEraser.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Jobs/Internal/JobTypeEraser.h>
+
+namespace AZ::Internal
+{
+    TypeErasedJob::TypeErasedJob(TypeErasedJob&& other) noexcept
+    {
+        if (!other.m_relocator || other.m_lambda != other.m_buffer)
+        {
+            // The type-erased lambda is trivially relocatable OR, the lambda is heap allocated
+            memcpy(this, &other, sizeof(TypeErasedJob));
+
+            if (other.m_lambda == other.m_buffer)
+            {
+                m_lambda = m_buffer;
+            }
+
+            // Prevent deletion in the event the lambda had spilled to the heap
+            other.m_lambda = nullptr;
+            return;
+        }
+
+        // At this point, we know the lambda was inlined
+        m_lambda = m_buffer;
+
+        m_invoker = other.m_invoker;
+        m_relocator = other.m_relocator;
+        m_destroyer = other.m_destroyer;
+
+        // We now own the lambda, so clear the moved-from job's destroyer
+        other.m_destroyer = nullptr;
+        other.m_invoker = nullptr;
+
+        m_relocator(m_buffer, other.m_buffer);
+    }
+
+    TypeErasedJob& TypeErasedJob::operator=(TypeErasedJob&& other) noexcept
+    {
+        if (this == &other)
+        {
+            return *this;
+        }
+
+        this->~TypeErasedJob();
+
+        new (this) TypeErasedJob{ AZStd::move(other) };
```

Comment:
This is one way to do it.
Not the best looking but does work.
Nitpick wise [std::destroy_at](https://en.cppreference.com/w/cpp/memory/destroy_at) and [std::construct_at](https://en.cppreference.com/w/cpp/memory/construct_at) could be used if you wanted to
```
AZStd::destroy_at(this);
AZStd::constuct_at(this, AZStd::move(other));
```
```

Reply:
Those are C++20 so I didn't know they were available. I don't know that `destroy_at` and `construct_at` are necessarily prettier, but don't really have any strong feelings so don't mind changing it

Reply:
That is why I mentioned it as a Nitpick. 
I am OK with not changing it.